omnigibson/utils/transform_utils.py

```diff
@@ -1326,6 +1326,43 @@ def random_quaternion(num_quaternions: int = 1) -> th.Tensor:
     return quaternions
 
 
+def compute_desired_pose_in_frame(prim, position, orientation, frame: Literal["world", "parent", "scene"] = "scene"):
```

Comment:
This function is hugely problematic: it does things for world and scene but leaves `local` as is, plus in its current form it would be much better as a function on BasePrim rather than here. I would prefer to remove this function actually. See my comments everywhere it's used for how to replace it.